components/list/list-item-drag-handle.js

```diff
@@ -240,6 +242,7 @@ class ListItemDragHandle extends LocalizeCoreElement(LitElement) {
 				@focusout="${this._onFocusOut}"
 				@keyup="${this._onActiveKeyboard}"
 				@keydown="${this._onPreventDefault}"
+				aria-live="assertive"
```

Comment:
This line does the magic. It tells the screen reader to read out the label if it changes, and IMMEDIATELY. 🎉